activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -141,6 +141,10 @@ def supports_datetime_with_precision?
         end
       end
 
+      def supports_virtual_columns?
+        !mariadb? && version >= "5.7.5"
```

Comment:
Thanks for the information. I should have read entire changes. 

Tested with MariaDB 10.2.2 beta and also I am unable to find virtual column expressions such as `'UPPER(name)'` from `information_schema.columns`.

```sql
MariaDB [yahonda]> CREATE TABLE `virtual_columns` (`id` bigint AUTO_INCREMENT PRIMARY KEY, `name` varchar(255), `upper_name` varchar(255) AS (UPPER(name)), `name_length` int AS (LENGTH(name)) PERSISTENT) ENGINE=InnoDB;
Query OK, 0 rows affected (0.05 sec)

MariaDB [yahonda]> show create table virtual_columns\G
*************************** 1. row ***************************
       Table: virtual_columns
Create Table: CREATE TABLE `virtual_columns` (
  `id` bigint(20) NOT NULL AUTO_INCREMENT,
  `name` varchar(255) DEFAULT NULL,
  `upper_name` varchar(255) AS (UPPER(name)) VIRTUAL,
  `name_length` int(11) AS (LENGTH(name)) PERSISTENT,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=latin1
1 row in set (0.00 sec)

MariaDB [yahonda]> select * from information_schema.columns
    -> where table_name = 'virtual_columns' and column_name in ('upper_name', 'name_length')\G
*************************** 1. row ***************************
           TABLE_CATALOG: def
            TABLE_SCHEMA: yahonda
              TABLE_NAME: virtual_columns
             COLUMN_NAME: upper_name
        ORDINAL_POSITION: 3
          COLUMN_DEFAULT: NULL
             IS_NULLABLE: YES
               DATA_TYPE: varchar
CHARACTER_MAXIMUM_LENGTH: 255
  CHARACTER_OCTET_LENGTH: 255
       NUMERIC_PRECISION: NULL
           NUMERIC_SCALE: NULL
      DATETIME_PRECISION: NULL
      CHARACTER_SET_NAME: latin1
          COLLATION_NAME: latin1_swedish_ci
             COLUMN_TYPE: varchar(255)
              COLUMN_KEY: 
                   EXTRA: VIRTUAL
              PRIVILEGES: select,insert,update,references
          COLUMN_COMMENT: 
*************************** 2. row ***************************
           TABLE_CATALOG: def
            TABLE_SCHEMA: yahonda
              TABLE_NAME: virtual_columns
             COLUMN_NAME: name_length
        ORDINAL_POSITION: 4
          COLUMN_DEFAULT: NULL
             IS_NULLABLE: YES
               DATA_TYPE: int
CHARACTER_MAXIMUM_LENGTH: NULL
  CHARACTER_OCTET_LENGTH: NULL
       NUMERIC_PRECISION: 10
           NUMERIC_SCALE: 0
      DATETIME_PRECISION: NULL
      CHARACTER_SET_NAME: NULL
          COLLATION_NAME: NULL
             COLUMN_TYPE: int(11)
              COLUMN_KEY: 
                   EXTRA: PERSISTENT
              PRIVILEGES: select,insert,update,references
          COLUMN_COMMENT: 
2 rows in set (0.00 sec)

MariaDB [yahonda]> select version();
+----------------+
| version()      |
+----------------+
| 10.2.2-MariaDB |
+----------------+
1 row in set (0.00 sec)

MariaDB [yahonda]>  
```

I understand the schema dumper implementation difficulties with MariaDB, still I think we need to have another name to switch schema dumping virtual column expressions or not.

Since it will be confusing users (like me) can create virtual columns even if `supports_virtual_columns?` returns `false`.  

How about `supports_dump_virtual_column_expressions?` ?